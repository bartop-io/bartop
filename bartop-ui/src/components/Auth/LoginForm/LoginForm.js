import React from 'react';
import PropTypes from 'prop-types';
import { Formik } from 'formik';

const LoginForm = ({ history, sendCode, prefillEmail }) => (
  <Formik
    initialValues={{
      email: prefillEmail || ''
    }}
    validate={({ email }) => {
      const errors = {};
      if (!email) {
        errors.email = 'Required';
      } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(email)) {
        errors.email = 'Invalid email address';
      }
      return errors;
    }}
    onSubmit={async ({ email }, { setSubmitting, setFieldError }) => {
      try {
        await sendCode(email);
        setSubmitting(false);
        history.replace({ pathname: '/auth/verify', state: { email } });
      } catch (err) {
        setSubmitting(false);
        setFieldError('email', err.message);
      }
    }}
    render={({
      values,
      errors,
      touched,
      handleChange,
      handleBlur,
      handleSubmit,
      isSubmitting
    }) => (
      <form onSubmit={handleSubmit}>
        <p>Enter your email to sign in or create an account</p>
        <input
          type="email"
          name="email"
          placeholder="bilbo@bartop.io"
          onChange={handleChange}
          onBlur={handleBlur}
          value={values.email}
        />
        {touched.email && errors.email && <div>{errors.email}</div>}
        <button type="submit" disabled={isSubmitting}>
          Submit
        </button>
      </form>
    )}
  />
);

LoginForm.propTypes = {
  history: PropTypes.object.isRequired,
  sendCode: PropTypes.func.isRequired,
  prefillEmail: PropTypes.string
};

export default LoginForm;
