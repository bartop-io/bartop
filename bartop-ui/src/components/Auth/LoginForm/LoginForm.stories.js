import React from 'react';

import { storiesOf } from '@storybook/react';

import LoginForm from './LoginForm';
import { noop } from '../../../test-helpers/utils';

const requiredProps = {
  history: {},
  sendCode: noop
};

storiesOf('LoginForm', module)
  .add('with required props', () => <LoginForm {...requiredProps} />)
  .add('with a prefillEmail', () => (
    <LoginForm {...requiredProps} prefillEmail="john@smith.com" />
  ));
