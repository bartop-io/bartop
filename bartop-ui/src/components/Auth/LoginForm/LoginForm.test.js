import React from 'react';
import { shallow } from 'enzyme';

import LoginForm from './LoginForm';

const requiredProps = {
  history: {},
  sendCode: jest.fn()
};

it('matches the snapshot', () => {
  const wrapper = shallow(<LoginForm {...requiredProps} />);
  expect(wrapper).toMatchSnapshot();
});
