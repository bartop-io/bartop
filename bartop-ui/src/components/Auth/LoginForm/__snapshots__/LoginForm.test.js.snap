// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<Formik
  enableReinitialize={false}
  initialValues={
    Object {
      "email": "",
    }
  }
  isInitialValid={false}
  onSubmit={[Function]}
  render={[Function]}
  validate={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
/>
`;
